### ISSUE TITLE
Enable FlexibleContexts by default?

### ISSUE THREAD

**Issue by lmbollen** [clash-lang/stack-templates#27]
When creating for example a small counter:

```hs
counter = register 0 ( counter + 1)
```
`counter` already requires a domain to be sampled. If you´d want to make this a counter for `System`.
You can´t make the type simply `Signal System (Unsigned 8)` because `register` requires `HiddenClockResetEnable dom`. 

For `System`, you now have to explicitly supply `HiddenClockResetEnable System`, which you can't because `FlexibleContexts` is not enabled by default. Now the user must also figure out what `LANGUAGE` pragmas are and how to set `FlexibleContexts`.

I think all these steps introduce a lot of friction and our template projects should maybe show how to deal with more of these situations.

**Comment by lmbollen** [comment]
I'm looking into the GHC2021 extensions, there are some we don't want such as `StarIsType` (we already have `NoStarIsType` enabled so that checks out) since using `StarIsType` breaks type level multiplication.

Furthermore using `NamedWildCards` causes errors in bittide without enabling `PartialTypeSignatures`. Specifically in combination with the `circuit-notation` plugin for circuits that are ignored. 

I don't think just enabling the GHC2021 blessed extensions is a great idea and we should just see what encounter and selectively enable extensions based on that.

A quick search in bittide shows that we enable the following extensions commonly on a per module basis:
```
RecordWildCards
NamedFieldPuns
FlexibleContexts
```

(I'm personally guilty of introducing most of those)
I'd prefer just adding those.

**Comment by lmbollen** [comment]
Solved by #33
